src/Worker/Core/Shims/TaskOrchestrationShim.cs

```diff
@@ -47,6 +47,7 @@ partial class TaskOrchestrationShim : TaskOrchestration
         object? input = this.DataConverter.Deserialize(rawInput, this.implementation.InputType);
         ILogger contextLogger = this.LoggerFactory.CreateLogger("Microsoft.DurableTask");
         this.wrapperContext = new(innerContext, this.invocationContext, contextLogger, input);
+        TaskOrchestrationContextAccessor.Current = this.wrapperContext;
```

Comment:
This should have been obvious to me, but the inner orchestrator is constructed before this point, so this fails. Will need to rethink how to smuggle the `TaskOrchestratorContext` for use in `OrchestrationLoggerFactory`.